hn-common/src/main/java/ca/bc/gov/hlth/hncommon/json/fhir/FHIRJsonUtil.java

```diff
@@ -108,22 +114,29 @@ private static void parseJsonArray(FHIRJsonMessage fhirJsonMsg, Object value) {
 				fhirJsonMsg.setV2MessageData(attachValue.toString());
 			} else {
 				logger.error("This is not an valid FHIR message!");
+				isParsed = false;
```

Comment:
Business logic is not same as old code. This change will process the for loop before returning false. In old business logic, it returned null first time only when an error is logged.